components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/autofill/LoginStorageDelegate.kt

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.autofill
+
+import androidx.annotation.VisibleForTesting
+import mozilla.appservices.logins.LoginsStorage
+import mozilla.components.concept.engine.Login
+import mozilla.components.lib.dataprotect.SecureAbove22Preferences
+import mozilla.components.service.sync.logins.ServerPassword
+import org.mozilla.geckoview.GeckoResult
+
+// Temporary Interface before lands in GV
+internal interface LoginDelegate {
+    fun onLoginUsed(login: Login)
+    fun onFetchLogins(domain: String): GeckoResult<Array<Login>>
+    fun onLoginSave(login: Login)
+}
+
+/**
+ * [LoginDelegate /* TODO UPDATE */] implementation.
+ *
+ * When GV needs to update or retrieve information about stored logins, it will call into this
+ * class.
+ *
+ * App will have to instantiate this and set it on the runtime and pass in the [LoginsStorage] and
+ * [SecureAbove22Preferences] with a key that conforms to "passwords".  This lets GV delegate to
+ * the client for storage.
+ */
+class LoginStorageDelegate(
+    private val loginStorage: LoginsStorage,
```

Comment:
@grigoryk should I be using the `AsyncLoginStorage` here? It looks like that's what you're doing in `FennecMigrator`? Also could you check the unlock/lock in these 3 funs pls? :)

Reply:
Per @grigoryk, we should be using `AsyncLoginStorage`, which is an AC wrapper around the AS class.